preprocessed.js

```diff
@@ -1,535 +0,0 @@
-
-      //number of numbers
-      rule(type('priceish'), score(numberOfNumbers)),
-
-      //number of dollar signs
-      rule(type('priceish'), score(numberOfDollarSigns)),
-
-      //number of decimal points
-      rule(type('priceish'), score(numberOfDots)),
-
-      //if in one of three formats: price range, price with decimal (2 numbers after decimal), price without decimal
-      rule(type('priceish'), score(priceFormat)),
-
-      //check meta tags with itemprop = price
-      rule(type('priceish'), score(metaTags)),
-
-      //return image with max score
-      rule(type('priceish').max(), out('product-price'))
-
-    );
-
-    function tuningRoutine(doc) {
-        return rules.against(doc).get('product-price');
-    }
-
-    return tuningRoutine;
-}
-
-/*
- * Maintain state as we compare a series of DOMs, reporting the percent difference at the end.
- */
-class DiffStats {
-    constructor(tuningRoutine, feature) {
-        this.feature = feature;
-        this.tuningRoutine = tuningRoutine || tuningRoutines[feature].routine;
-    }
-
-    compare(coeffs) {
-        let gotText;
-        if (this.feature === 'image') {
-          gotText = withoutQueryParams(this.tuningRoutine(...coeffs)(window.document).map(fnode => fnode.element.src)[0]);
-
-        } else if (this.feature === 'title') {
-          //compare innerHTML text of titles
-          gotText = this.tuningRoutine(...coeffs)(window.document).map(fnode => fnode.element.innerHTML)[0];
-
-        } else if (this.feature === 'price') {
-          //strip whitespace, dollar sign, words, and trailing zeros when comparing price
-          gotText = this.tuningRoutine(...coeffs)(window.document).map(fnode => fnode.element)[0];
-          gotText = formatPrice((gotText.tagName !== 'META')? gotText.textContent : gotText.getAttribute('content'));
-
-        }
-        this.gotText = gotText;
-    }
-
-    result() {
-        return this.gotText;
-    }
-}
-
-/*
- * Calculate overall score for one feature
- * @param {string} feature title/image/price
- * @param {object} array of tuning coeffs if any
- */
-function deviationScore(feature, coeffs = []) {
-    const stats = new DiffStats(tuningRoutines[feature].routine, feature);
-    stats.compare(coeffs);
-    return stats.result();
-}
-
-console.log(deviationScore('title', tuningRoutines['title'].coeffs));
-console.log(deviationScore('image', tuningRoutines['image'].coeffs));
-console.log(deviationScore('price', tuningRoutines['price'].coeffs));
-
-
-// Inform the background page that
-// this tab should have a page-action
-browser.runtime.sendMessage({
-  from:    'content',
-  subject: 'showPageAction'
-});
-
-// Listen for messages from the popup
-browser.runtime.onMessage.addListener(function (msg, sender, response) {
-  // First, validate the message's structure
-  if ((msg.from === 'popup') && (msg.subject === 'DOMInfo')) {
-    var domInfo = {
-      title:  deviationScore('title', tuningRoutines['title'].coeffs),
```

Comment:
I'm confused by the use of deviationScore here, which is a tuning routine. I would have expected something like `tunedImageFnodes().against(document)`, which would invoke the already-tuned image-finding ruleset using the default coefficients.